sdk/messaging/azservicebus/message_batch.go

```diff
@@ -150,6 +152,26 @@ func (mb *MessageBatch) addAMQPMessage(tempMsg amqpCompatibleMessage) error {
 	return nil
 }
 
+func (mb *MessageBatch) injectCreationCtx(ctx context.Context, tracer tracing.Tracer, sendSpan tracing.Span) error {
+	mb.mu.Lock()
+	defer mb.mu.Unlock()
+
+	for i, msg := range mb.marshaledMessages {
```

Comment:
I'll look at this some more - I'd have for the act of adding telemetry to force us to deseralize and then reserialize every message again.

Reply:
In fact, can we just add this instrumentation code into MessageBatch.AddMessage() instead, so each message gets instrumented as they are added?

Reply:
That would require us to take in a `context.Context` from the user at `AddMessage()` and therefore changing a public facing API.

But I agree the deseralize -> reserialize part is a bit ugly, if you are down to change the API signature there, we can definitely add it in `AddMessage()`

Reply:
Good point. I think we're going to need to rethink this because you can't add stuff to a message _after_ it's been added to a batch, because it increases the size, which can lead to a valid batch being rejected when sent since now it's grown past it's initial size.

This one we might want to leave out, and I can add it back in when I come up with a solution. We don't need to hold the entire PR for it.

Reply:
Currently I kept the tracing for it but removed the context propagation. So we'd still have a `send` trace, just the message doesn't have an injected context.